gadget/gadget.go

```diff
@@ -771,6 +823,28 @@ func orderStructuresByOffset(vss []VolumeStructure) []VolumeStructure {
 	return ordVss
 }
 
+func validatePartial(v *Volume) error {
+	var foundCache [len(validPartialProperties)]bool
+	for _, p := range v.Partial {
+		found := false
+		for vpvIdx, valid := range validPartialProperties {
+			if p == valid {
+				if foundCache[vpvIdx] == true {
+					return fmt.Errorf("partial value %q is repeated", p)
+				}
+				foundCache[vpvIdx] = true
+				found = true
+				break
+			}
+		}
+		if found == false {
```

Comment:
(super nitpick) I slightly prefer `if !found {` as it reads in my head as `if not found` which is slightly shorter than `if found is false' but this is probably the mother of all nitpicks :)